### PR TITLE
Add Thread Payloads Parsing

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Relates to: https://github.com/GetStream/ios-issues-tracking/issues/752

### 🎯 Goal
Adds thread payload parsing to the SDK. This is part of the Threads V2 implementation. The goal is to separate the implementation into multiple PRs so that it is easier to review.

### 🧪 Manual Testing Notes
N/A. Only internal code was added, and it is all covered by unit tests.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json [hunk 1]
```diff
@@ -0,0 +1,2400 @@
+{
+  "channel_cid": "messaging:4AB11F2F-4",
```
Comment (martinmitrevski) [inline TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json:2]: I don't think it's necessary to have such large JSONs for this. Can you reduce them to few entries?

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ThreadListPayload.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct ThreadListPayload: Decodable {
```
Comment (martinmitrevski) [inline Sources/StreamChat/APIClient/Endpoints/Payloads/ThreadListPayload.swift:7]: Why don't we also add the networking part in this PR?

File: TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json (hunk 1 above)
Comment (nuno-vieira) [reply TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json:2]: Yeah I copy and pasted from a real response, but I can reduce it 👍

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ThreadListPayload.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/ThreadListPayload.swift:7]: It is just easier to review if the PRs are split into smaller ones

File: TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json (hunk 1 above)
Comment (nuno-vieira) [reply TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json:2]: Done ✅

**Review by laevandus — APPROVED** [review]

File: TestTools/StreamChatTestTools/Fixtures/JSONs/ThreadList.json [hunk 3]
```diff
@@ -0,0 +1,5072 @@
...
+            "created_at": "2024-03-26T11:13:35.162494Z",
+            "updated_at": "2024-03-26T11:13:35.162494Z",
+            "banned": false,
+            "shadow_banned": false,
+            "role": "owner",
+            "channel_role": "channel_member",
+            "notifications_muted": false
+          },
+          {
+            "user_id": "tommaso-0",
```
Comment (martinmitrevski) [inline TestTools/StreamChatTestTools/Fixtures/JSONs/ThreadList.json:83]: There are so many users here, we don't really need this, we just need couple of these.

File: TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json [hunk 4]
```diff
@@ -0,0 +1,2400 @@
...
+        "created_at": "2024-03-26T11:13:35.162494Z",
+        "updated_at": "2024-03-26T11:13:35.162494Z",
+        "banned": false,
+        "shadow_banned": false,
+        "role": "owner",
+        "channel_role": "channel_member",
+        "notifications_muted": false
+      },
+      {
+        "user_id": "tommaso-0",
```
Comment (martinmitrevski) [inline TestTools/StreamChatTestTools/Fixtures/JSONs/Thread.json:81]: Same here, this JSON can be still reduced.

**Review by martinmitrevski — COMMENTED** [review]
Let's aim for JSONs less than <1000 lines, there's no need for all that data there.

File: TestTools/StreamChatTestTools/Fixtures/JSONs/ThreadList.json (hunk 3 above)
Comment (nuno-vieira) [reply TestTools/StreamChatTestTools/Fixtures/JSONs/ThreadList.json:83]: Done

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**